cpp_linter/clang_tools/__init__.py

```diff
@@ -46,6 +46,18 @@ def _run_on_single_file(
 ) -> Tuple[str, str, Optional[TidyAdvice], Optional[FormatAdvice]]:
     log_stream = worker_log_init(log_lvl)
 
+    format_advice = None
+    if format_cmd is not None and (
+        format_filter is None or format_filter.is_source_or_ignored(file.name)
+    ):
+        format_advice = run_clang_format(
+            command=format_cmd,
+            file_obj=file,
+            style=args.style,
+            lines_changed_only=args.lines_changed_only,
+            format_review=args.format_review,
+        )
```

Comment:
Moved running clang-format before tidy in case tidy fails to write back the unmodified file then atleast format has had a chance to run on the original source